### PR TITLE
Tines Rule - Story Items Destruction

### PR DESCRIPTION
**Pull request by josh-panther**
### Background

https://www.tines.com/api/audit-logs#:~:text=StoryItemsDestruction

### Changes

* Tines - StoryItemsDestruction rule

### Testing

```
Tines.Story.Items.Destruction
	[PASS] Detection Trigger
		[PASS] [rule] true
		[PASS] [title] Tines [StoryItemsDestruction] performed by [user@email.com] on [tines-log-source-name].
		[PASS] [dedup] Tines [StoryItemsDestruction] performed by [user@email.com] on [tines-log-source-name].
		[PASS] [alertContext] {"actor": "user@email.com", "action": "StoryItemsDestruction", "tenant_id": "1337", "user_email": "user@email.com", "user_id": "7331", "operation_name": "StoryItemsDestruction", "request_ip": "98.224.225.84"}
	[PASS] Tines Login
		[PASS] [rule] false
```

### CODE REVIEW

**Comment by andrea-youwakim** [comment]
failing lint and test

File: rules/tines_rules/tines_story_items_destruction.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def rule(event):
+    return event.get('operation_name') == 'StoryItemsDestruction'
```
Comment (edyesed) [inline rules/tines_rules/tines_story_items_destruction.py:2]: please use `deep_get()` with a `default=` kwarg in place of `event.get()`

File: rules/tines_rules/tines_story_items_destruction.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+def title(event):
+    operation = event.get('operation_name')
+    user = event.get('user_email')
+    tines_instance = event.get('p_source_label')
+    return f"Tines {operation} by {user} on {tines_instance}"
```
Comment (edyesed) [inline rules/tines_rules/tines_story_items_destruction.py:8]: Variable interpolation should use square brackets around the interpolation. please convert these to also use `deep_get`
```suggestion
    return f"Tines [{operation}] by [{user}] on [{tines_instance}]"
```

Comment (edyesed) [inline rules/tines_rules/tines_story_items_destruction.py:8]: This detection should use the global helper for tines to provide alert_context https://github.com/panther-labs/panther-analysis/blob/master/global_helpers/panther_tines_helpers.py#L1

File: rules/tines_rules/tines_story_items_destruction.py (hunk 1 above)
Comment (edyesed) [reply rules/tines_rules/tines_story_items_destruction.py:2]: The Tines.Audit log type has a global filter, this should be the first thing that happens inside the rule function https://github.com/panther-labs/panther-analysis/blob/master/rules/tines_rules/tines_custom_ca.py#L11-L12

File: rules/tines_rules/tines_story_items_destruction.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+AnalysisType: rule
+DisplayName: "Tines Story Items Destruction"
+Enabled: true
+Filename: tines_story_items_destruction.py
+Severity: Low
+Tests:
```
Comment (edyesed) [inline rules/tines_rules/tines_story_items_destruction.yml:6]: We need a test case that has `ExpectedResult: false` to demonstrate that this does not alert on every log event passed into it.

**Review by edyesed — CHANGES_REQUESTED** [review]

**Comment by josh-panther** [comment]
Waiting for https://github.com/panther-labs/panther-analysis/pull/786 before review req

**Comment by josh-panther** [comment]
I made this info level severity per https://www.notion.so/Hackathon-Tines-StoryActions-Destruction-a7a2da9e783e47e1aab1ceea9d3fef6f?d=ee9596cd6dfc4bfdaea23f1b8130cf7d&pvs=4

If someone wants to override me on that feel free but I don't feel super comfortable with it even at "Low" - Given my understanding of how Tines workflows typically go. If I'm wrong just let me know 👍

**Review by edyesed — APPROVED** [review]
adds a Info level detection for Tines story items destruction. While this is a data-destruction classified event, it is also a somewhat common event in Tines stories as workflows are modified.

**Pull request merged**